code/game/machinery/vending.dm

```diff
@@ -949,6 +963,37 @@ var/global/num_vending_terminals = 1
 			new_price = R.price
 		R.price = new_price
 
+	else if (href_list["delete_entry"] && src.vend_ready && !currently_vending)
```

Comment:
From what I can see, it's just that the button to delete entries does not appear unless amount == 0.
However it can come to happen that the button somehow gets pressed anyways, such as from 2 people using the machine at once or someone having an outdated window etc
If that happens, the game will run this code block anyways, even if the button wasn't supposed to be press-able.

You should add a check in this code block that verifies that amount == 0 just in case. Hrefs are tricky like that.

Reply:
Good point. I've added edit_mode and some amount/length checking to the code now.